### ISSUE TITLE
Allure report no longer shows which steps had errors after `Narrator` update.

### ISSUE THREAD

**Issue by pgoy-dod** [perrygoy/screenpy#74]
There's some weirdness with the new Adapter approach:
* Exceptions are no longer thrown in a way which looks like the step errored. The step says it passed!
   * before: 
![Screen Shot 2021-08-30 at 12 28 22 PM](https://user-images.githubusercontent.com/57111606/131379854-435c5eda-1a8e-4510-b7e4-2ab727a480cb.png)
  * after: 
![Screen Shot 2021-08-30 at 12 28 43 PM](https://user-images.githubusercontent.com/57111606/131379894-7113f376-7880-4eac-a0bd-295143a54845.png)
* Because neither `Eventually` nor `See` have `beat`s, the exceptions raised  are not linked to the Action and the full time to execute `Eventually` is not shown.
  * example: 
![Screen Shot 2021-08-30 at 12 30 13 PM](https://user-images.githubusercontent.com/57111606/131380173-e96eb495-232d-4666-a634-1f456330e358.png)

Is there anything we can do to fix these problems?

**Comment by perrygoy** [comment]
So, the actionable item that i *can* take from this is figuring out how to make `Eventually` and `See` have loggable statements.

I've half-started and abandoned this approach several times before when `See` was the only Action on the block who didn't have a `beat`. The approach i think i'm going to take is to add a `__repr__` or `__str__` definition to each action, which will describe in present-tense essentially what their call looked like. For example:

```pycon
>>> WELCOME_BANNER = Target.the("Welcome Banner").located_by("banner#welcome")
>>> click_action = Click.on_the(WELCOME_BANNER)
>>> print(click_action)
Click on the Welcome Banner.
```

`See` and `Eventually` will both do their own work to figure out how to log their stored Actions, Questions, etc.

**Comment by perrygoy** [comment]
I just realized i left off a little bit at the beginning of my last comment—i can't figure out how to persist the contexts through exceptions being raised. I tried doing the same trick i did for the `StdOutAdapter`, i tried switching from yielding inside contextmanagers to wrapping... nothing i tried worked.

Maybe someone else has an idea for how to address those issues, but for now i'm going to rename this issue to focus on `Eventually` and `See` not having `beat`s.

**Comment by perrygoy** [comment]
Unfortunately, adding beats did not magically encapsulate the errors for Allure, which is a bummer.

I'm going to make a separate issue for the beats part and keep this one about the weird Allure report behavior open.

**Comment by pgoy-dod** [comment]
I have an idea!

I have a different reason for suggesting that the `Narrator` have a method like `explains_an_error`, which would broadcast an exception to the various adapters. This might be the trick to making Allure work again! 

It all depends on what trappings Allure itself uses to hook an exception to a `step`. We'll probably have to do the same in the `AllureAdapter` to get it to work.

**Comment by perrygoy** [comment]
Tadaaa, fixed in [d4e5d02](https://github.com/perrygoy/screenpy/commit/d4e5d02647fbda923a9404ef8eeee2344ad5495f)!
